src/perl/lib/wtsi_clarity/epp/generic/stamper.pm

```diff
@@ -31,6 +32,52 @@ has 'step_url' => (
   required   => 1,
 );
 
+override 'run' => sub {
+  my $self = shift;
+  super(); #call parent's run method
+  $self->epp_log('Step url is ' . $self->step_url);
+
+  if ($self->shadow_plate && $self->copy_on_target ) {
+    croak qq{One cannot use the shadow plate stamping with the copy_on_target option!};
+  }
+
+  ##no critic ValuesAndExpressions::ProhibitMagicNumbers
+  if ($self->group && $self->input_artifacts->findnodes('/art:details/art:artifact')->size() > 96) {
+    croak q{Can not do a group stamp for more than 96 inputs!};
+  }
+  ##use critic
+
+  $self->_create_containers();
+  my $doc = $self->_create_placements_doc;
+
+  if ($self->copy_on_target) {
```

Comment:
What should happen if copy_on_target and group flags both are true? Is copy_on_target is the stronger flag or a validation is missing for this case?